### PR TITLE
DOCSP-40792: Delete guide

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-40792
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/cpp-driver/DOCSP-40792-delete/write/delete/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/write/delete.txt [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    :language: cpp
+    :dedent:
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+Delete Operations
+-----------------
+
+You can perform delete operations in MongoDB by using the following methods:
```
Comment (jordan-smith721) [inline source/write/delete.txt:50]:
```suggestion
You can perform delete operations with the {+driver-short+} by using the following methods:
```

File: source/write/delete.txt [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+- ``delete_one()``, which deletes *the first document* that matches the search criteria
+- ``delete_many()``, which deletes *all documents* that match the search criteria
+
+Each delete method requires a **query filter** document, which specifies the
+search criteria that determine which documents to select for removal.
```
Comment (jordan-smith721) [inline source/write/delete.txt:56]: S:
```suggestion
Each delete method requires a **query filter** document, which specifies the
search criteria that is used to determine which documents to select for removal.
```

File: source/write/delete.txt [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+         fields. For more information, see the :manual:`let statement
+         </reference/command/delete/#std-label-delete-let-syntax>` in the
+         {+mdb-server+} manual.
+
+   * - ``comment``
+     - | Attaches a comment to the operation. For more information, see the :manual:`delete command
+         fields </reference/command/delete/#command-fields>` guide in the
+         {+mdb-server+} manual.
+
+The following code uses the ``delete_many()`` method to delete all documents in
```
Comment (jordan-smith721) [inline source/write/delete.txt:127]:
```suggestion
The following example calls the ``delete_many()`` method to delete all documents in
```

File: source/write/delete.txt [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+- ``result()``, which returns the raw bulk write result
+- ``deleted_count()``, which returns the number of documents deleted
+
+If the query filter does not match any documents, the driver doesn't delete any
+documents and ``deleted_count`` is 0.
+
+The following example uses the ``delete_many()`` method to delete documents
+that have a ``cuisine`` value of ``"Greek"`` and prints the number of deleted
+documents:
```
Comment (jordan-smith721) [inline source/write/delete.txt:159]: S: Split this to sentence and point out that the `deleted_count()` function is being used
```suggestion
The following example calls the ``delete_many()`` method to delete documents
that have a ``cuisine`` value of ``"Greek"``. It then calls the ``deleted_count()`` member function to print the number of deleted
documents:
```

**Review by jordan-smith721 — APPROVED** [review]
LGTM w/ a few suggestions

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**